### PR TITLE
refactor: Split Java and ObjC code into Reanimated and Worklets dirs

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

This PR is the continuation of the code split made in common C++ layer. All of the files currently go to `reanimated` directory, moving some of them to `worklets` will happen in follow-up PRs.

Requires #6357.

## Test plan

- [ ] CI passes.

### CODE REVIEW

File: packages/react-native-reanimated/android/build.gradle [hunk 1]
```diff
@@ -430,15 +430,16 @@ def assertMinimalReactNativeVersion = task assertMinimalReactNativeVersionTask {
...
 
 task prepareHeadersForPrefab(type: Copy) {
-    from("$projectDir/src/main/cpp")
+    from("$projectDir/src/main/cpp/reanimated")
     from("$projectDir/../Common/cpp/reanimated/AnimatedSensor")
     from("$projectDir/../Common/cpp/reanimated/Fabric")
     from("$projectDir/../Common/cpp/reanimated/LayoutAnimations")
     from("$projectDir/../Common/cpp/reanimated/NativeModules")
+    from("$projectDir/../Common/cpp/reanimated/Tools")
+    from("$projectDir/../Common/cpp/worklets/ReanimatedRuntime")
```
Comment (tomekzaw) [inline packages/react-native-reanimated/android/build.gradle:439]: something is no yes

File: packages/react-native-reanimated/android/build.gradle [hunk 2]
```diff
@@ -349,63 +349,63 @@ android {
     sourceSets.main {
         java {
             if (IS_NEW_ARCHITECTURE_ENABLED) {
-                srcDirs += "src/fabric/java"
+                srcDirs += "src/fabric/reanimated/java"
```
Comment (tomekzaw) [inline packages/react-native-reanimated/android/build.gradle:352]: I don't think we need `reanimated/` here since we're about to introduce a new separate `build.gradle` for worklets

File: packages/react-native-reanimated/android/src/main/cpp/reanimated/CMakeLists.txt [hunk 3]
```diff
@@ -1,14 +1,14 @@
 cmake_minimum_required(VERSION 3.8)
 
 file(GLOB_RECURSE REANIMATED_COMMON_SOURCES CONFIGURE_DEPENDS "${REANIMATED_COMMON_DIR}/*.cpp")
-file(GLOB_RECURSE SOURCES_ANDROID CONFIGURE_DEPENDS "${ANDROID_SRC_DIR}/main/cpp/*.cpp")
+file(GLOB_RECURSE SOURCES_ANDROID CONFIGURE_DEPENDS "${REANIMATED_ANDROID_DIR}/*.cpp")
```
Comment (tomekzaw) [inline packages/react-native-reanimated/android/src/main/cpp/reanimated/CMakeLists.txt:4]: this needs to be cleaned up since `REANIMATED_ANDROID_DIR` might be confusing (at first glance it looks like it points to `packages/react-native-reanimated/android`)

File: packages/react-native-reanimated/apple/reanimated/REAModule.mm [hunk 4]
```diff
@@ -326,6 +326,7 @@ - (void)commonInit:(std::shared_ptr<NativeReanimatedModule>)nativeReanimatedModu
      withRnRuntime:(jsi::Runtime &)rnRuntime
 {
   WorkletRuntimeCollector::install(rnRuntime);
+
```
Comment (tomekzaw) [inline packages/react-native-reanimated/apple/reanimated/REAModule.mm:329]: mmhm

File: packages/react-native-reanimated/apple/reanimated/native/NativeProxy.mm [hunk 5]
```diff
@@ -83,7 +82,7 @@ static inline bool getIsReducedMotion()
       platformDepMethodsHolder,
       valueUnpackerCode,
       isBridgeless,
-      isReducedMotion);
+      getIsReducedMotion());
```
Comment (tomekzaw) [inline packages/react-native-reanimated/apple/reanimated/native/NativeProxy.mm:85]: plz move to separate PR

**Review by tomekzaw — CHANGES_REQUESTED** [review]

File: packages/react-native-reanimated/apple/reanimated/native/NativeProxy.mm (hunk 5 above)
Comment (tjzel) [reply packages/react-native-reanimated/apple/reanimated/native/NativeProxy.mm:85]: plz no

**Review by tomekzaw — APPROVED** [review]

**Review by piaskowyk — APPROVED** [review]

**Review by tomekzaw — APPROVED** [review]
